fiboa_cli/publish.py

```diff
@@ -0,0 +1,97 @@
+    """
+    assert dataset in list_all_converter_ids()
+    os.makedirs(directory, exist_ok=True)
+
+    for key in ("README.md", "LICENSE.txt"):
+        path = os.path.join(directory, key)
+        if not os.path.exists(path):
+            raise FileNotFoundError(f"Please create {path}")
+
+    parent = os.path.dirname(directory)
+    os.chdir(parent)
+
+    if not source_coop_extension:
+        source_coop_extension = dataset.replace("_", "-")
+
+    file_name = source_coop_extension.replace("-", "_")  # not sure if we want this
+
+    parquet_file = os.path.join(directory, f"{file_name}.parquet")
+    source_coop_url = f"https://beta.source.coop/fiboa/{source_coop_extension}/"
+
+    assert requests.get(f"https://api.source.coop/repositories/fiboa/{source_coop_extension}").status_code == 200, \
+        f"Missing repo at {source_coop_url}"
+
+    collection_file = os.path.join(directory, "collection.json")
+
+    stac_directory = os.path.join(directory, "stac")
+    done_convert = os.path.exists(parquet_file) and os.path.exists(os.path.join(stac_directory, "collection.json"))
+
+    if not done_convert:
+        # fiboa convert xx_yy -o data/xx-yy.parquet -h https://beta.source.coop/fiboa/xx-yy/ --collection
+        log(f"Converting file for {dataset} to {parquet_file}\n", "success")
+        convert(dataset, parquet_file, cache, source_coop_url, collection=True)
+        log(f"Done\n", "success")
+    else:
+        log(f"Using existing file {parquet_file} for {dataset}\n", "success")
+
+    # fiboa validate data/xx-yy.parquet --data
+    log(f"Validating {parquet_file}", "info")
+    result = validate(parquet_file, config={"data": True})
+    if result:
+        log("\n  => VALID\n", "success")
+    else:
+        log("\n  => INVALID\n", "error")
+        sys.exit(1)
+
+    # mkdir data/stac; mv data/collection.json data/stac
+    stac_target = os.path.join(stac_directory, "collection.json")
+    if not done_convert:
+        os.makedirs(stac_directory, exist_ok=True)
+        data = json.load(open(collection_file))
+        assert data["id"] == dataset, f"Wrong collection dataset id: {data['id']} != {dataset}"
+
+        if STAC_EXTENSION not in data["stac_extensions"]:
+            data["stac_extensions"].append(STAC_EXTENSION)
+            data["links"].append({
+                "href": f"{source_coop_url}/{file_name}.pmtiles",
+                "type": "application/vnd.pmtiles",
+                "rel": "pmtiles"
+            })
+
+        with open(stac_target, "w", encoding="utf-8") as f:
+            json.dump(data, f)
+        os.remove(collection_file)
+
+    pm_file = os.path.join(directory, f"{file_name}.pmtiles")
+    if not os.path.exists(pm_file):
+        log(f"ogr2ogr geo.json", "info")
+        exc(f"ogr2ogr -t_srs EPSG:4326 geo.json {parquet_file}")
+
+        log(f"Running tippecanoe", "info")
+        exc(f"tippecanoe -zg --projection=EPSG:4326 -o {pm_file} -l {dataset} geo.json --drop-densest-as-needed")
```

Comment:
This has some heavy dependencies that would need to be installed somehow. Running into GDAL installation issues everytime, I feel like this might be hard to solve properly...

Reply:
Maybe have these ones be a little more 'opt-in'? Like we just check if the libraries are installed, and if not it just says 'please install GDAL and tippecanoe', and skips the steps.

Reply:
Thanks for the valid points in the review. I realized this pull-request was a work-in-progress that "works on my machine", but wanted to check if it was an interesting feature for the project. 

To get a data-set published/updated, you need to follow [the HOWTO](https://github.com/fiboa/data/blob/main/HOWTO.md#each-time-you-update-the-dataset) and you will need gdal & tippecanoe on your machine. I relied on `os.system()` calls as opposed to python-libraries to avoid introducing more dependencies. Whenever I see a list of detailed instructions like in the HOWTO, I'm always afraid I (or one of my project-members) will mess up / forget something. A script like this -with enough guidance and sanity checks- will keep us all safe. Let's see if I can get some more sanity checks in there...

Reply:
I think it's a good idea. I think this might need to be a bit more interactive, giving instructions to users and waiting for confirmation, ... - more like a step-by-step wizard ;-)

Reply:
Ok. @m-mohr I'm not sure what part you're considering making interactive. We could pre-generate the README and LICENSE by scraping the specific page from the [data-survey](https://raw.githubusercontent.com/fiboa/data-survey/main/data/FR.md). Or we could request a copy/paste-input of secrets from source coop. 

Currently the script stops with some instructions (write README, install tippecanoe, copy the `export AWS_` commands in the terminal.

Reply:
I think he meant just meant interactive with the requirements for the 'additional things' - like prompting someone 'would you like to generate pmtiles y/n?', and then have it check if tippecanoe is installed, and if not tell the user 'tippecanoe must be installed to generate pmtiles, do XYZ to install it'.